streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,8 +1081,9 @@ private void shutdown(final boolean cleanRun) {
         streamsMetrics.removeAllSensors();
     }
 
+    // visible for testing
     @SuppressWarnings("ThrowableNotThrown")
-    private void shutdownTasksAndState(final boolean cleanRun) {
+    void shutdownTasksAndState(final boolean cleanRun) {
```

Comment:
Could we use reflection on such not-only-test functions: http://tutorials.jenkov.com/java-reflection/private-fields-and-methods.html ?